src/pybind/mgr/dashboard/services/rgw_client.py

```diff
@@ -1179,51 +1123,79 @@ def get_updated_endpoints(endpoints):
                 if user_details:
                     keys = user_details['keys'][0]
+                    access_key = keys['access_key']
+                    secret_key = keys['secret_key']
+                    logger.info("access_key: %s, secret_key: %s", access_key, secret_key)
                     logger.info("Modifying zone with user credentials: %s", username)
-                    self.modify_zone(zone_name=zone_name, zonegroup_name=zonegroup_name,
-                                     default='true', master='true', endpoints=zone_ip_url,
-                                     access_key=keys['access_key'],
-                                     secret_key=keys['secret_key'])
+                    if access_key and secret_key:
+                        rgw_multisite_instance.modify_zone(zone_name=zone_name,
+                                                           zonegroup_name=zonegroup_name,
+                                                           default='true', master='true',
+                                                           endpoints=zone_ip_url,
+                                                           access_key=keys['access_key'],
+                                                           secret_key=keys['secret_key'])
+                    else:
+                        raise ValueError("Access key or secret key is missing")
         except Exception as e:
             logger.error("Failed to create zone or system user: %s", e)
+            mgr.remote('progress', 'fail', self.progress_id, str(e))
             raise
         try:
             # Restart RGW daemons and set credentials
             logger.info("Restarting RGW daemons and setting credentials")
+            self.update_progress("Restarting RGW daemons and setting credentials")
             rgw_service_manager = RgwServiceManager()
             rgw_service_manager.restart_rgw_daemons_and_set_credentials()
+            self.progress_done += 1
         except Exception as e:
             logger.error("Failed to restart RGW daemons: %s", e)
+            mgr.remote('progress', 'fail', self.progress_id, str(e))
             raise
         try:
             # Get realm tokens and import to another cluster if specified
             logger.info("Getting realm tokens")
             realm_token_info = CephService.get_realm_tokens()
+            logger.info("Realm tokens: %s", realm_token_info)
 
             if cluster_fsid and realm_token_info:
                 logger.info("Importing realm token to cluster: %s", cluster_fsid)
+                self.update_progress(f"Importing realm token to cluster: {cluster_fsid}")
                 self.import_realm_token_to_cluster(cluster_fsid, realm_name,
                                                    realm_token_info, username)
+            else:
+                self.update_progress("Realm Export Token fetched successfully")
+                mgr.remote('progress', 'complete', self.progress_id)
```

Comment:
Instead of setting these two seperately, we can just pass `progress title` and `progress action` in `update_progress` function

Reply:
Thanks, updated the code